ledger/participant-integration-api/src/main/scala/platform/store/backend/common/Field.scala

```diff
@@ -91,11 +92,21 @@ private[backend] case class BooleanOptional[FROM](extract: FROM => Option[Boolea
   override def convert: Option[Boolean] => lang.Boolean = _.map(Boolean.box).orNull
 }
 
+private[backend] object Timestamp {
+  def instantToMicros(i: Instant): Long =
+    TimeUnit.SECONDS.toMicros(i.getEpochSecond) + TimeUnit.NANOSECONDS.toMicros(i.getNano.toLong)
```

Comment:
ChronoUnit.MICROS.between(Instant.EPOCH, i)

Reply:
This code is ~copied from~ inspired by `com.daml.lf.data.Time.Timestamp` (which I did not want to depend on).

What would be the advantage of `ChronoUnit`? `ChronoUnit.MICROS.between` executes more code than `TimeUnit.SECONDS.toMicros`.

Reply:
I am not sure what you mean...I had only the client usage in mind, that seems simpler then summing up a tomicros getEpochSecond and a toMicros getNano getLong, or not?